R/XcmsExperiment-functions.R

```diff
@@ -1016,3 +1030,71 @@ featureArea <- function(object, mzmin = min, mzmax = max, rtmin = min,
         .require_spectra()
         Spectra::Spectra(res)
     }
+
+#' helper function to get indices of chromatographic peaks for eventual
+#' subsetting with `rt`, `mz`, `msLevel` and `type`.
+#'
+#' @param object `XcmsExperiment` or `XCMSnExp` object with `chromPeaks`.
+#'
+#' @return `integer` vector with the indices of the `chromPeaks` matching the
+#'     requested filtering.
+#'
+#' @noRd
+.index_chrom_peaks <- function(object, rt = numeric(),
```

Comment:
this function is then also re-used in `chromPeaks,XcmsExperiment` and `chromPeaks,XCMSnExp`.